### PR TITLE
Adding tests to 'xr curl' command

### PR DESCRIPTION
**Pull request by gailazar300**
- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray_test.go [hunk 1]
```diff
@@ -174,3 +174,18 @@ func verifyScanResults(t *testing.T, content string, minViolations, minVulnerabi
...
+	err := xrayCli.WithoutCredentials().Exec("xr", "curl", "-XGET", "/api/v1/system/version")
+	assert.NoError(t, err)
+	// Configure a new server named "default".
+	createJfrogHomeConfig(t, true)
+	// Check curl command with '--server-id' flag
+	err = xrayCli.WithoutCredentials().Exec("xr", "curl", "-XGET", "/api/system/version", "--server-id=default")
+	assert.NoError(t, err)
+	// Check curl command with invalid server id - should get an error.
+	err = xrayCli.WithoutCredentials().Exec("xr", "curl", "-XGET", "/api/system/version", "--server-id=not_configured_name")
+	assert.Error(t, err)
```
Comment (yahavi) [inline xray_test.go:190]:
```suggestion
	assert.EqualError(t, err, "Server ID 'not_configured_name' does not exist.")
```

File: xray_test.go [hunk 2]
```diff
@@ -174,3 +174,18 @@ func verifyScanResults(t *testing.T, content string, minViolations, minVulnerabi
 	assert.True(t, len(results[0].Vulnerabilities) >= minVulnerabilities, fmt.Sprintf("Expected at least %d vulnerabilities in scan results, but got %d vulnerabilities.", minVulnerabilities, len(results[0].Vulnerabilities)))
 	assert.True(t, len(results[0].Licenses) >= minLicenses, fmt.Sprintf("Expected at least %d Licenses in scan results, but got %d Licenses.", minLicenses, len(results[0].Licenses)))
 }
+
+func TestXrayCurl(t *testing.T) {
+	initXrayTest(t, commands.GraphScanMinXrayVersion)
```
Comment (yahavi) [inline xray_test.go:179]: I'm not sure we need the version check, can we just check this?
```go
if !*tests.TestXray {
	t.Skip("Skipping Xray test. To run Xray test add the '-test.xray=true' option.")
}
```
For example, maybe in `initXrayVersion` we can return early when no min Xray version is provided as a parameter.

**Review by yahavi — CHANGES_REQUESTED** [review]
Looks good! 😎 
Please also add `CleanXrayTests` in main_test.go at`tearDownIntegrationTests()`.
This cleanup method should do the same as `cleanBuildToolsTest()`:
```go
os.Unsetenv(coreutils.HomeDir)
tests.CleanFileSystem()
```
 Maybe we can rename `cleanBuildToolsTest()` to something more generic and use it.

File: main_test.go [hunk 3]
```diff
@@ -115,8 +115,8 @@ func prepareHomeDir(t *testing.T) (string, string) {
 	return oldHomeDir, newHomeDir
 }
 
-func cleanBuildToolsTest() {
-	if *tests.TestNpm || *tests.TestGradle || *tests.TestMaven || *tests.TestGo || *tests.TestNuget || *tests.TestPip || *tests.TestDocker {
+func cleanTestsHomeEnv() {
+	if *tests.TestNpm || *tests.TestGradle || *tests.TestMaven || *tests.TestGo || *tests.TestNuget || *tests.TestPip || *tests.TestDocker || *tests.TestXray {
```
Comment (eyalbe4) [inline main_test.go:119]: I'm not sure this if statement is actually needed.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**